src/colmap/util/threading_ctrl.h

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2023, ETH Zurich and UNC Chapel Hill.
```

Comment:
Maybe rename this file (including `ControllerThread`) to `controllers/base_controller.h`?

Reply:
This is done. But do we want to put ControllerThread elsewhere?

Reply:
I think that it can stay there for now